### PR TITLE
changes for error handling and decreasing sdk size

### PR DESCRIPTION
**Pull request by ruchiramoitra**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/411)

### CODE REVIEW

File: analytics.js [hunk 1]
```diff
@@ -176,26 +176,27 @@ class Analytics {
...
+          // else we do not consider any filtering for cookie consent.
+          this.clientIntegrations = this.clientIntegrations.filter((intg) => {
+            return (
+              !cookieConsent || // check if cookieconsent object is present and then do filtering
+              (cookieConsent && cookieConsent.isEnabled(intg.config))
+            );
+          });
+        } catch (e) {
+          logger.error(e);
+        }
```
Comment (saikumarrs) [inline analytics.js:197]: Why do we have to put the entire thing under try-catch? Aren't we just concerned about initializing the consent object

File: cookieConsent/OneTrust/browser.js [hunk 2]
```diff
@@ -1,15 +1,12 @@
...
   constructor() {
     // If user does not load onetrust sdk before loading rudderstack sdk
     // we will not be filtering any of the destinations.
     if (!window.OneTrust || !window.OnetrustActiveGroups) {
-      // logger.debug(
-      //   `Onetrust window objects not retrieved. Thus events are sent.`
-      // );
-      return true;
+      throw new Error(
+        "Onetrust window objects not retrieved. Thus events are sent."
```
Comment (saikumarrs) [inline cookieConsent/OneTrust/browser.js:8]:
```suggestion
        "OneTrust resources are not accessible. Thus all the destinations will be loaded"
```

File: integrationBuildScript.js [hunk 3]
```diff
@@ -15,7 +15,7 @@ intgNamesArr.forEach((intgName) => {
       `npm run buildProdIntegrationCLI --intg=${intgName}`,
       { encoding: "utf-8" }
     );
-    logger.debug("Done!");
+    // logger.debug("Done!");
```
Comment (saikumarrs) [inline integrationBuildScript.js:18]: This is not part of the SDK package. Is this change needed?

File: cookieConsent/OneTrust/browser.js [hunk 4]
```diff
@@ -77,7 +74,6 @@ class OneTrust {
 
       return containsAllConsent;
     } catch (e) {
-      logger.error(`Error during onetrust cookie consent management ${e}`);
```
Comment (saikumarrs) [inline cookieConsent/OneTrust/browser.js:80]: This is necessary. Let's keep it

File: analytics.js (hunk 1 above)
Comment (ruchiramoitra) [reply analytics.js:197]: Then even if it goes to catch it will be executed

File: cookieConsent/OneTrust/browser.js (hunk 4 above)
Comment (ruchiramoitra) [reply cookieConsent/OneTrust/browser.js:80]: Size increasing because of this.

Comment (saikumarrs) [reply cookieConsent/OneTrust/browser.js:80]: Can you find something else to optimize? Worst case we'll increase the limit but cannot compromise on errors

File: analytics.js (hunk 1 above)
Comment (saikumarrs) [reply analytics.js:197]: Got it

**Review by saikumarrs — APPROVED** [review]

**Comment by ruchiramoitra** [comment]
no longer needed
